main.go

```diff
@@ -219,3 +221,9 @@ func processEvents(ctx context.Context, logger hclog.Logger, extensionClient *ex
 		}
 	}
 }
+
+// userAgent returns a user-agent-style string
+func userAgent() string {
+	// the DevEx team uses UAs that look like "vault-client-go/0.0.1 (Darwin arm64; Go go1.19.2)"
+	return fmt.Sprintf("%s/%s (%s %s; Go %s)", extensionName, extensionVersion, runtime.GOOS, runtime.GOARCH, runtime.Version())
```

Comment:
I think we need to also specify whether the secret is being requested in order to write to the tmp file, or requested via the proxy server.

Reply:
that's a good catch, the issue does state this as part of the acceptance criteria:
> User-Agent header specifies whether the secret is being requested in order to write to the tmp file, or requested via the proxy server.

Thinking about this, perhaps my background in the VLE is not extensive enough, is this kind of information appropriate to the user-agent header? It almost sounds characteristic of the request, rather than characteristic of the requestor.